types/node/index.d.ts

```diff
@@ -2367,40 +2377,44 @@ declare module "url" {
         unicode?: boolean;
     }
 
-    export class URLSearchParams implements Iterable<[string, string]> {
-        constructor(init?: URLSearchParams | string | { [key: string]: string | string[] | undefined } | Iterable<[string, string]> | Array<[string, string]>);
-        append(name: string, value: string): void;
-        delete(name: string): void;
-        entries(): IterableIterator<[string, string]>;
-        forEach(callback: (value: string, name: string) => void): void;
-        get(name: string): string | null;
-        getAll(name: string): string[];
-        has(name: string): boolean;
-        keys(): IterableIterator<string>;
-        set(name: string, value: string): void;
-        sort(): void;
-        toString(): string;
-        values(): IterableIterator<string>;
-        [Symbol.iterator](): IterableIterator<[string, string]>;
-    }
-
-    export class URL {
-        constructor(input: string, base?: string | URL);
-        hash: string;
-        host: string;
-        hostname: string;
-        href: string;
-        readonly origin: string;
-        password: string;
-        pathname: string;
-        port: string;
-        protocol: string;
-        search: string;
-        readonly searchParams: URLSearchParams;
-        username: string;
-        toString(): string;
-        toJSON(): string;
+    global {
```

Comment:
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/25342

I'm working on build failures caused by 10.0.0 being pulled in as a transitive dependency and I wonder whether the decision to move these definitions into the global namespace is causing the collision here.

Reply:
It was definitely made a global because `node` now places it into the global scope (which is problematic if node is in the same compilation as the `dom` lib), this really needs https://github.com/Microsoft/TypeScript/pull/15780 (and for the `URL` constructor to be in its own lib) for it to be handled well.

Reply:
@weswigham I'm apprehensive that even #15780 will help since the WHATWG URL implementation has a subtly different API surface in the DOM vs. NodeJS. Rather, I think this will necessitate switching the definition of `URL` and `URLSearchParams` in DOM to use the same split interface (instance vs. constructor) approach we use for Object, Number, etc. so that the declarations can merge.

Reply:
I'm reverting this specific change in #25356 for now.

Reply:
I believe this same type of problem exists for `setTimeout`.

The implementation is subtly different in the DOM vs Node.js, specifically the return type. https://nodejs.org/dist/latest-v10.x/docs/api/timers.html

Somehow TypeScript is able to handle this scenario, right?